protocol/bc/legacy/map.go

```diff
@@ -143,17 +145,10 @@ func mapTx(tx *TxData) (headerID bc.Hash, hdr *bc.TxHeader, entryMap map[bc.Hash
 
 			var (
 				anchorID    bc.Hash
-				setAnchored func(*bc.Hash)
+				setAnchored = func(*bc.Hash) {}
```

Comment:
This feels a little hackish to me. Slight preference for testing `setAnchored != nil` at the point where it's called.